GNUmakefile

```
@@ -353,22 +358,25 @@ else
 	@go test -v ./agent -run Vault
 endif
 
-.PHONY: protoc-check
-protoc-check:
-	$(info checking protocol buffer compiler version (expect: $(PROTOC_VERSION)))
-	@if ! command -v protoc &>/dev/null; then \
-		echo "ERROR: protoc is not installed; please install version $(PROTOC_VERSION)" >&2 ; \
-		exit 1 ; \
-	fi
-	@if [[ "$$(protoc --version | cut -d' ' -f2)" != "$(PROTOC_VERSION)" ]]; then \
-		echo "ERROR: protoc version $(PROTOC_VERSION) is required" >&2 ; \
+.PHONY: protoc-install
+protoc-install:
+	$(info locally installing protocol buffer compiler version if needed (expect: $(PROTOC_VERSION)))
+	@if [[ ! -x $(PROTOC_ROOT)/bin/protoc ]]; then \
+		mkdir -p .protobuf/tmp ; \
+		if [[ ! -f .protobuf/tmp/$(PROTOC_ZIP) ]]; then \
+			( cd .protobuf/tmp && wget $(PROTOC_URL) ) ; \
+		fi ; \
+		mkdir -p $(PROTOC_ROOT) ; \
+		unzip -d $(PROTOC_ROOT) .protobuf/tmp/$(PROTOC_ZIP) ; \
```

Won't this fail in the case where we have to download `protoc` because it'll have `cd`'d into `.protobuf/tmp` above on 367?

```suggestion
		unzip -d $(PROTOC_ROOT) $(PROTOC_ZIP) ; \
```

---

Nope. The `cd` happens in a subshell (the parentheses) so after the closing parenthesis the working directory is back to the original.